portal/import_template_ui.php

```diff
@@ -15,303 +15,543 @@
+                        <div class="form-group" id="category_group">
+                            <label class="font-weight-bold mx-1" for="template_category"><?php echo xlt('Category'); ?></label>
+                            <select class="form-control" id="template_category" name="template_category">
+                                <?php echo $select_cat_options ?>
+                            </select>
+                        </div>
+                    </div>
+                    <div class='btn-group ml-1'>
+                        <button type='submit' class='btn btn-search btn-secondary'><?php /*echo xlt('Refresh'); */ ?></button>
+                        <button type='button' id="send-button" class='btn btn-transmit btn-outline-primary d-none' onclick="return sendTemplate()">
+                            <?php echo xlt('Send'); ?>
+                        </button>
+                        <button type='button' id="upload-nav-button" name='upload-nav-button' class='btn btn-primary' data-toggle='collapse' data-target='#upload-nav'>
+                            <?php echo xlt('Files') ?>
+                        </button>
                     </div>
-                    <button type="submit" class="btn btn-secondary"><?php echo xlt('Refresh'); ?></button>
+                    <div class="ml-auto">
+                        <label class="form-check"><?php echo xlt('Use Popout Editor'); ?>
+                            <input type='checkbox' class='form-check-inline mx-1' id='is_modal' name='is_modal' checked='checked' />
+                        </label>
+                    </div>
+                    <input type='hidden' id='upload-nav-value' name='upload-nav-value' value='<?php echo $_REQUEST['upload-nav-value'] ?? 'hidden' ?>' />
                 </form>
+            </nav>
+            <!-- Upload -->
+            <nav class="collapse my-2 <?php echo $_REQUEST['upload-nav-value'] ?>" id="upload-nav">
+                <div class='col col-12'>
+                    <form id='form_upload' class='form-inline row' action='import_template.php' method='post' enctype='multipart/form-data'>
+                        <hr />
+                        <div class='col'>
+                            <div id='upload_scope_category'></div>
+                            <div class='mb-2' id='upload_scope'></div>
+                        </div>
+                        <div class='form-group col'>
+                            <div class='form-group'>
+                                <input type='file' class='btn btn-outline-info' id="fetch_files" name='template_files[]' multiple />
+                                <button class='btn btn-outline-success d-none' type='submit' name='upload_submit' id='upload_submit'><i class='fa fa-upload' aria-hidden='true'></i></button>
+                            </div>
+                        </div>
+                        <input type='hidden' name='upload_pid' value='<?php echo json_encode($patient); ?>' />
```

Comment:
this is an interesting one. I think would should also wrap this with attr()

Reply:
it's an array

Reply:
Makes sense to convert array to json. json_encode won't protect it in the html context and this variable is brought in via a request (ie. risk of reflected xss). Wrapping the json_encode with attr() in this setting `attr(json_encode($patient))` should not break anything (since the escaping will be unraveled when then form is submitted via html).